python/pyspark/sql/catalog.py

```diff
@@ -138,6 +139,78 @@ def listColumns(self, tableName, dbName=None):
                 isBucket=jcolumn.isBucket()))
         return columns
 
+    @ignore_unicode_prefix
+    @since(3.0)
+    def databaseExists(self, dbName):
+        """Check if the database with the specified name exists."""
+        return self._jcatalog.databaseExists(dbName)
+
+    @ignore_unicode_prefix
+    @since(3.0)
+    def functionExists(self, functionName, dbName=None):
+        """Check if the function with the specified name exists.
+
+        If no database is specified, the current database is used.
+        """
+        if dbName is None:
+            dbName = self.currentDatabase()
+        return self._jcatalog.functionExists(dbName, functionName)
+
+    @ignore_unicode_prefix
+    @since(3.0)
+    def tableExists(self, tableName, dbName=None):
```

Comment:
ditto for matching signautre.

Reply:
Same comment, I don't think matching the Scala API is necessarily the right thing to do here. I think this more closely matches Python developers expectations of positional arguments behaviour.